### PR TITLE
feat: ETH connector migration.

### PR DESCRIPTION
**Pull request by paouvrard**

### CODE REVIEW

File: packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts [hunk 1]
```diff
@@ -598,35 +606,26 @@ export async function checkBurn (
...
+  } else {
+    // Withdraw ERC-20 ETH from a silo which doesn't use ETH as native currency.
+    // @ts-expect-error
+    const txBlock = await nearProvider.block({ blockId: nearBurnTx.transaction_outcome.block_hash })
+    const blockHeight = Number(txBlock.header.height)
+    const etherNep141FactoryMigrationHeight = options.etherNep141FactoryMigrationHeight ?? bridgeParams.etherNep141FactoryMigrationHeight
+    const etherNep141Factory = blockHeight >= etherNep141FactoryMigrationHeight
+      ? (options.etherNep141Factory ?? bridgeParams.etherNep141Factory)
+      : bridgeParams.auroraEvmAccount
+    nearBurnReceipt = await parseETHBurnReceipt(nearBurnTx, etherNep141Factory, nearProvider)
```
Comment (olga24912) [inline packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts:621]: The same code is used in the recover function. Can we extract it into a separate function?

File: packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts [hunk 2]
```diff
@@ -598,35 +606,26 @@ export async function checkBurn (
...
-      nearBurnReceipt = await parseETHBurnReceipt(nearBurnTx, transfer.auroraEvmAccount ?? 'aurora', nearProvider)
-    }
-  } catch (e) {
-    if (e instanceof TransferError) {
-      return {
-        ...transfer,
-        errors: [...transfer.errors, e.message],
-        status: status.FAILED
-      }
-    }
```
Comment (olga24912) [inline packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts:616]: Why was this error handling removed?

File: packages/aurora-ether/src/bridged-ether/sendToEthereum/index.ts [hunk 3]
```diff
@@ -246,6 +246,8 @@ export async function recover (
   sender: string = 'todo',
   options?: TransferOptions & {
     nep141Factory?: string
+    etherNep141Factory?: string
+    etherNep141FactoryMigrationHeight?: number
```
Comment (olga24912) [inline packages/aurora-ether/src/bridged-ether/sendToEthereum/index.ts:250]: TransferOptions already contains this args

File: packages/near-ether/src/bridged-ether/sendToEthereum/index.ts [hunk 4]
```diff
@@ -704,9 +702,13 @@ export async function checkSync (
     options.nearAccount?.connection.provider ??
     getNearProvider()
   if (nearOnEthClientBlockHeight > withdrawBlockHeight) {
+    const etherNep141FactoryMigrationHeight = options.etherNep141FactoryMigrationHeight ?? bridgeParams.etherNep141FactoryMigrationHeight
+    const etherNep141Factory = withdrawBlockHeight >= etherNep141FactoryMigrationHeight
+      ? (options.etherNep141Factory ?? bridgeParams.etherNep141Factory)
+      : bridgeParams.auroraEvmAccount
```
Comment (olga24912) [inline packages/near-ether/src/bridged-ether/sendToEthereum/index.ts:708]: This code is duplicated many times. Can it be extracted?

**Review by olga24912 — DISMISSED** [review]

File: packages/near-ether/src/bridged-ether/sendToEthereum/index.ts (hunk 4 above)
Comment (paouvrard) [reply packages/near-ether/src/bridged-ether/sendToEthereum/index.ts:711]: Yea good idea. https://github.com/aurora-is-near/rainbow-bridge-client/pull/98/commits/a49efb543ac2aa0ea430a994b940bdf50181e4cd

File: packages/aurora-ether/src/bridged-ether/sendToEthereum/index.ts (hunk 3 above)
Comment (paouvrard) [reply packages/aurora-ether/src/bridged-ether/sendToEthereum/index.ts]: https://github.com/aurora-is-near/rainbow-bridge-client/pull/98/commits/a49efb543ac2aa0ea430a994b940bdf50181e4cd

File: packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts (hunk 2 above)
Comment (paouvrard) [reply packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts:616]: Cleaning up an error that doesn't exist anymore, if it fails checkStatus loop will try again.

File: packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts [hunk 5]
```diff
@@ -261,6 +261,8 @@ export async function recover (
   options?: TransferOptions & {
     decimals?: number
     symbol?: string
+    etherNep141Factory?: string
+    etherNep141FactoryMigrationHeight?: number
```
[the remark this answers is not in the document]
Comment (paouvrard) [reply packages/aurora-erc20/src/bridged-erc20/sendToEthereum/index.ts]: https://github.com/aurora-is-near/rainbow-bridge-client/pull/98/commits/a49efb543ac2aa0ea430a994b940bdf50181e4cd

**Review by karim-en — APPROVED** [review]

**Review by olga24912 — DISMISSED** [review]

**Review by olga24912 — APPROVED** [review]

**Comment by frol** [comment]
@paouvrard Sorry for jumping in, but I'd like to invite you to participate in the [Race of Sloths](https://race-of-sloths.com) - a fun and gamified open-source contributions program. Consider mentioning @race-of-sloths user in your github comment or PR description to join!

See how the flow works here: https://github.com/near/nearcore/pull/11778

**Review by olga24912 — APPROVED** [review]

**Pull request merged**